src/modules/returns/lib/jobs/init-upload.js

```diff
@@ -9,12 +10,18 @@ const registerSubscribers = async messageQueue => {
   await messageQueue.subscribe(jobs.mapToJson.jobName, jobs.mapToJson.handler);
 
   await messageQueue.onComplete(jobs.start.jobName, async job => {
-    // XML document has been validated against XSD schema.
-    //
-    // Publish a new job which will convert the valid XML into
-    // a JSON blob which will be uploaded back to S3.
-    const { eventId } = job.data.request.data;
-    await jobs.mapToJson.publish(eventId);
+    if (job.data.failed) return messageQueue.stop();
```

Comment:
The `messageQueue` is a singleton instance for the whole application so I don't think it should be stopped.
Possibly all we would wish to do here is set the event to an `error` status and return before further processing.

Reply:
I wasn't sure what I should be doing here, that's why I was keen someone with pg-boss experience had a look at this